cumulusci/tests/pytest_plugins/pytest_sf_vcr.py

```diff
@@ -7,36 +7,117 @@
-        r"//.*.my.salesforce.com", "//orgname.my.salesforce.com", request.uri
-    )
-    request.uri = re.sub(
-        r"//.*\d+.*.salesforce.com/", "//orgname.my.salesforce.com/", request.uri
-    )
-    request.uri = re.sub(r"00D[\w\d]{15,18}", "Organization/ORGID", request.uri)
+replacements = [
+    (r"/v?\d\d.0/", r"/vxx.0/"),
+    (r"/00D[\w\d]{12,15}", "/00D0xORGID00000000"),
+    (r'"00D[\w\d]{12,15}"', '"00D0xORGID00000000"'),
+    (r".com//", r".com/"),
+    (r"ersion>\d\d.0<", r"ersion>vxx.0<"),
+    (r"<sessionId>.*</sessionId>", r"<sessionId>**Elided**</sessionId>"),
+    (r"001[\w\d]{12,15}", "0010xACCOUNTID0000"),
+    (r"//.*.my.salesforce.com", "//orgname.my.salesforce.com"),
+    (r"//.*\d+.*.salesforce.com/", "//orgname.my.salesforce.com/"),
+    (
+        r"202\d-\d\d-\d\dT\d\d:\d\d\:\d\d.\d\d\d\+0000",
+        "2021-01-01T01:02:01.000+0000",
+    ),
+    (r'"005[\w\d]{12,15}"', '"0050xUSERID0000000"'),
+    (r'"InstanceName" : "[A-Z]{2,4}\d{1,4}",', '"InstanceName" : "CS420",'),
+    (r"<id>0.*<\/id>", lambda m: "<id>0ANID</id>"),
+    (
+        r"<asyncProcessId>0.*<\/asyncProcessId>",
+        lambda m: "<asyncProcessId>0ANAPPID</asyncProcessId>",
+    ),
+    (r"\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d.\d\d\dZ", "2021-01-03T01:11:11.420Z"),
+    (r"/User/005[\w\d]{12,15}", "/User/0050xUSERID0000000"),
+    (r"<lastModifiedById>005[\w\d]{12,15}", "<lastModifiedById>005USERID"),
+]
+
+replacements = [
+    (re.compile(pattern), replacement) for (pattern, replacement) in replacements
+]
+
+
+RECORDING = ContextVar("RECORDING", default=True)
+READING = ContextVar("READING", default=True)
+
+
+@contextmanager
+def set_mode(mode, value):
+    orig = mode.get()
+    mode.set(value)
+    try:
+        yield
+    finally:
+        mode.set(orig)
+
+
+@pytest.fixture(scope="function")
+def vcr_cassette_path(vcr_cassette_dir, vcr_cassette_name):
+    return Path(vcr_cassette_dir, vcr_cassette_name + ".yaml")
+
+
+@pytest.fixture(autouse=True)
+def configure_recording_mode(
+    request,
+    user_requested_network_access,
+    vcr_cassette_path,
+    user_requested_cassette_replacement,
+):
+    recording_mode = RECORDING.get()
+    reading_mode = READING.get()
+    if (
+        user_requested_network_access
+        and vcr_cassette_path.exists()
+        and user_requested_cassette_replacement
+    ):
+        vcr_cassette_path.unlink()
+        recording_mode = True
+        reading_mode = "irrelevant"  # reading from missing file
+    elif user_requested_network_access and vcr_cassette_path.exists():
+        # user wants to keep existing cassette, so disable VCR usage entirely, like:
+        # https://github.com/ktosiek/pytest-vcr/blob/08482cf0724697c14b63ad17752a0f13f7670add/pytest_vcr.py#L56
+        recording_mode = False
+        reading_mode = False
+    elif user_requested_network_access:
+        recording_mode = True
+        reading_mode = "irrelevant"  # file doesn't exist, so reading missing file
```

Comment:
I don't love that these values can be either arbitrary strings or Booleans. How about enums?